server/frontendserver.go

```diff
@@ -2,43 +2,62 @@ package server
 
 import (
 	"log"
+	"net"
+	"net/http"
 	"os"
 
+	"github.com/gorilla/rpc"
+	"github.com/gorilla/rpc/json"
 	"github.com/privacylab/talek/common"
 )
 
-// FrontendServer is an RPC server for a Frontend API
+// FrontendServer represents an HTTP served Frontend
 type FrontendServer struct {
-	log     *log.Logger
-	name    string
-	rpcPort int
+	log  *log.Logger
+	name string
 
-	frontend *Frontend
-	netRPC   *NetworkRPC
-	netHTTP  *NetworkHTTP
+	Frontend *Frontend
+	*rpc.Server
 }
 
-// NewFrontendServer creates a new FrontEndServer for a given configuration.
-func NewFrontendServer(name string, rpcPort int, serverConfig *Config, replicas []common.TrustDomainConfig) *FrontendServer {
+// NewFrontendServer creates a new Frontend implementing HTTP.Handler.
+func NewFrontendServer(name string, serverConfig *Config, replicas []*common.TrustDomainConfig) *FrontendServer {
 	fe := &FrontendServer{}
 	fe.log = log.New(os.Stdout, "[FrontendServer:"+name+"] ", log.Ldate|log.Ltime|log.Lshortfile)
 	fe.name = name
-	fe.rpcPort = rpcPort
 
 	rpcs := make([]common.ReplicaInterface, len(replicas))
 	for i, r := range replicas {
-		rpcs[i] = common.NewReplicaRPC(r.Name, &r)
+		rpcs[i] = common.NewReplicaRPC(r.Name, r)
 	}
 
-	fe.frontend = NewFrontend(name, serverConfig, rpcs)
-	if rpcPort != 0 {
-		fe.netRPC = NewNetworkRPC(fe.frontend, rpcPort)
-	}
-	//fe.netHttp = NewNetworkHttp(httpPort)
+	fe.Frontend = NewFrontend(name, serverConfig, rpcs)
+
+	// Set up the RPC server component.
+	fe.Server = rpc.NewServer()
+	fe.Server.RegisterCodec(&json.Codec{}, "application/json")
+	fe.Server.RegisterTCPService(fe.Frontend, "Frontend")
+
 	return fe
 }
 
-// Kill stops a frontend server.
-func (fe *FrontendServer) Kill() {
-	fe.netRPC.Kill()
+// Run begins an HTTP server for the server at a specific address
+func (fe *FrontendServer) Run(address string) (net.Listener, error) {
```

Comment:
Coordinator server doesn't have this Run function.
Can we be consistent?

Also, where does the listener get closed?

Reply:
frontendserver was the rpc server for the frontend. there hasn't been an equivalent RPC server for the coordinator yet, and i'm not actually sure what the set of configuration is where that fits in at this point to feel like i could add it appropriately